src/implementation/Client/GRPCClient/state.ts

```diff
@@ -158,14 +174,21 @@ export default class GRPCClientState implements IClientState {
     });
   }
 
-  async delete(storeName: string, key: string): Promise<void> {
+  async delete(storeName: string, key: string, options?: StateDeleteOptions): Promise<StateSaveResponseType> {
     const msgService = new DeleteStateRequest();
     msgService.setStoreName(storeName);
     msgService.setKey(key);
 
-    // @todo: implement below
-    // msgService.setEtag();
-    // msgService.setOptions();
+    if (options?.etag) {
+      const etag = new Etag();
+      etag.setValue(options.etag);
+      msgService.setEtag(etag);
+    }
+
+    const stateOptions = new StateOptions();
+    if (options?.consistency) stateOptions.setConsistency(options.consistency as any);
```

Comment:
Also multi line ifs as per previous

Maybe also easy to introduce a helper method instead of duplicated code?

E.g., `_configureStateOptions();`